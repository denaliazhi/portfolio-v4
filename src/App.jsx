import { useState } from "react";
import Sample from "./components/Sample";
import "./App.css";

function App() {
  const [heading, setHeading] = useState("Sprinkles");
  return (
    <>
      <h1>{heading}</h1>
      <button
        onClick={() => {
          setHeading("Chocolate");
        }}
      >
        Change
      </button>

      <Sample></Sample>
    </>
  );
}

export default App;
