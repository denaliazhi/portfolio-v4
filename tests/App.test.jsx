import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "../src/App";

describe("App", () => {
  it("renders 'Sprinkles'", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders 'Chocolate'", async () => {
    const user = userEvent.setup();

    render(<App />);
    const button = screen.getByRole("button", { name: "Change" });
    await user.click(button);
    screen.debug();

    expect(screen.getByRole("heading").textContent).toMatch(/Chocolate/i);
  });
});
