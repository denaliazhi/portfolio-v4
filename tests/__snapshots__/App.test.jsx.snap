// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders 'Sprinkles' 1`] = `
<div>
  <h1>
    Sprinkles
  </h1>
  <button>
    Change
  </button>
</div>
`;
